### ISSUE TITLE
Use Relational Database instead of NoSQL

### ISSUE THREAD

**Issue by stjohnjohnson** [screwdriver-cd/screwdriver#399]
Right now we're using DynamoDB to store our data.  It worked well for us at first, but we've encountered two things that make it hard to move forward:
 - Our IDs are hashes, which are hard to remember.  _"Oh yeah, project 1fec866296c698141c2e8e0ed21da0ba23c848f1"_  Implementing auto-incrementing IDs in NoSQL is difficult.
 - Pagination doesn't work as one would expect (no way to do page X, limit Y).

Instead, we're going to move our system to use relational databases (like PostgreSQL) and switch to numeric IDs.

---

- [x] Update Data Schema
- [x] Update Models
- [x] Update Sequelize
- [x] Remove DynamoDB
- [x] Fix all the API stuff
- [x] Delete Everything
- [x] Recreate all the things

**Comment by stjohnjohnson** [comment]
EOD Summary:
 - Made changes to Data Schema to use new ID
 - Found out that keeping Hash was complicated and didn't provide any value
 - Started upgrade on Datastore Base and Sequelize
 - Working on change to model to support get by ID OR unique keys

**Comment by stjohnjohnson** [comment]
Right now we're just waiting on the API rollout: https://github.com/screwdriver-cd/screwdriver/pull/401

**Comment by tkyi** [comment]
Disabled prod job for API and pointed beta job to new postgres DB. Merged #401 

Currently debugging acceptance test failure to create pipelines.

**Comment by tkyi** [comment]
Found a bunch of bugs:
- [ ] Need to change datastore-sequelize to use `upsert` instead of `create`. Dynamodb allowed for overwriting of table entries, postgres does not. (Seeing a SequelizeUniqueConstraintError)
- [ ] Need to update sdstore docker image. Seeing validation errors `\"Build ID\" fails because [\"Build ID\" length must be 40 characters long]"`
- [ ] Need to investigate error. Seeing `Error: Failed to create job: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"jobs.batch \"10\" already exists","reason":"AlreadyExists","details":{"name":"10","group":"batch","kind":"jobs"},"code":409}`

**Comment by stjohnjohnson** [comment]
@d2lam for the upsert, why are we creating the same thing twice?  Shouldn't it be updates? https://github.com/screwdriver-cd/datastore-sequelize/blob/master/index.js#L281

For the second one, use v2.0.0: https://github.com/screwdriver-cd/store/pull/15

For the third, we may need to start prefixing the k8s job names.

**Comment by d2lam** [comment]
@stjohnjohnson 
1.  From what we see, it gets the error when it gets to [pull request sync](https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/webhooks/index.js#L170). Inside startPRJob, it tries to [create a new job again](https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/webhooks/index.js#L70), which will eventually call `save`. For sync, this job already exists, so we get to this unique constraint error. 

startPRJob is used by pullRequestOpened as well, so it makes sense it creates a new job there. I suppose if we don't want to go this route, we can split this up. If it's PR opened, then create a new job. If not, then update.

2.  We couldn't try updating to v2.0.0 because wouldn't that affect the production also? Didn't see an option to update store image for beta only.

**Comment by stjohnjohnson** [comment]
1. Interesting.  We should fix that logic to lookup first.
2. Looks like there isn't a store-beta: https://github.com/screwdriver-cd/kubernetes/blob/master/store.yaml

**Comment by tkyi** [comment]
Seeing an interesting error after merging #408 
```bash
shoulderwork-lm:screwdriver tkyi$ kubectl describe jobs 1
Name:		1
Namespace:	default
Image(s):	node:6
Selector:	controller-uid=b75e0d50-d21a-11e6-b5f6-023b52e01fad
Parallelism:	1
Completions:	1
Start Time:	Tue, 03 Jan 2017 17:11:26 -0800
Labels:		app=screwdriver
		sdbuild=1
		tier=builds
Pods Statuses:	0 Running / 1 Succeeded / 0 Failed
Volumes:
  screwdriver:
    Type:	EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:	
Events:
  FirstSeen	LastSeen	Count	From			SubobjectPath	Type		Reason			Message
  ---------	--------	-----	----			-------------	--------	------			-------
  14m		14m		1	{job-controller }			Normal		SuccessfulCreate	Created pod: 1-yzer4
```
```bash
shoulderwork-lm:screwdriver tkyi$ kubectl logs -f 1-yzer4 build
2017/01/04 01:11:28 Processing logs for build 1
2017/01/04 01:11:28 Archiver started
2017/01/04 01:11:28 Starting Build 1
2017/01/04 01:11:28 Setting Build Status to RUNNING
2017/01/04 01:11:29 Fetching Build 1
2017/01/04 01:11:29 Error running launcher: fetching Build ID 1: 404 Not Found: Build does not exist
2017/01/04 01:11:29 Setting build status to FAILURE
2017/01/04 01:11:29 Archiver stopped
2017/01/04 01:11:29 Processing complete for build 1
2017/01/04 01:11:29 Failed updating the build status: posting to Build Status: 404 Not Found: Build 1 does not exist
```

All tables begin with index `2`.
<img width="512" alt="screen shot 2017-01-03 at 5 33 48 pm" src="https://cloud.githubusercontent.com/assets/3230529/21629196/db72133c-d1da-11e6-963d-6e2eff9c2c12.png">
<img width="708" alt="screen shot 2017-01-03 at 5 34 50 pm" src="https://cloud.githubusercontent.com/assets/3230529/21629208/fce845c2-d1da-11e6-8943-9609216ac626.png">

**Comment by d2lam** [comment]
Hm probably not related to #408

**Comment by tkyi** [comment]
The beta deploy job's functional tests ran successfully, verifying that the recent PRs should be working.

**Next steps:**
- [ ] Run prod deploy job
- [ ] Manually change API deployment to use sequelize vs dynamodb
- [ ] Recreate API pipeline
- [ ] Rerun prod deploy job
- [ ] Recreate the rest of the pipelines
